Source/Core/DolphinQt/Debugger/MemoryWidget.cpp

```diff
@@ -533,6 +542,56 @@ void MemoryWidget::OnSetValue()
   Update();
 }
 
+void MemoryWidget::OnSetValueFromFile()
+{
+  if (!Core::IsRunning())
+    return;
+
+  bool good_address, good_offset;
+  // Returns 0 if conversion fails
+  u32 addr = m_search_address->text().toUInt(&good_address, 16);
+  addr += m_search_offset->text().toUInt(&good_offset, 16);
+
+  if (!good_address)
+  {
+    ModalMessageBox::critical(this, tr("Error"), tr("Bad address provided."));
+    return;
+  }
+
+  if (!good_offset)
+  {
+    ModalMessageBox::critical(this, tr("Error"), tr("Bad offset provided."));
+    return;
+  }
+
+  QString path = QFileDialog::getOpenFileName(this, tr("Select a file"), QDir::currentPath(),
+                                              QStringLiteral("All files (*.*)"));
+  if (path.isNull())
+  {
+    ModalMessageBox::critical(this, tr("Error"), tr("No file provided."));
+    return;
+  }
+
+  File::IOFile f(path.toStdString(), "rb");
+
+  if (!f)
+  {
+    ModalMessageBox::critical(nullptr, QObject::tr("Error"), QObject::tr("Unable to open file."));
```

Comment:
This line is weird, why the `nullptr` and `QObject::`? Should probably be just `ModalMessageBox::critical(this, ("Error"), tr("Unable to open file."));`